File: Java/프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.LinkedList; //import
import java.util.Queue; //import

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "";
        Queue<String> deck1 = new LinkedList<>();
        Queue<String> deck2 = new LinkedList<>();
        for(int i=0;i<cards1.length;i++){
            deck1.add(cards1[i]);
        }
        for(int i=0;i<cards2.length;i++){
            deck2.add(cards2[i]);
        }
        
        for(int i=0;i<goal.length;i++){
            if(goal[i].equals(deck1.peek())){
                deck1.poll();
            }
            else if(goal[i].equals(deck2.peek())){
                deck2.poll();
            }
            else{
                return "No";
            }
        }
        
        return "Yes";
    }
}

File: Kotlin/프로그래머스/1/159994. 카드 뭉치/카드 뭉치.kt
class Solution {
    fun solution(cards1: Array<String>, cards2: Array<String>, goal: Array<String>): String {
        var answer: String = "Yes"
        var cards1Index = 0
        var cards2Index = 0
        goal.forEachIndexed{i,v -> 
            if(cards1Index<=cards1.lastIndex && cards1[cards1Index] == goal[i]) {
                cards1Index++
            } else if(cards2Index<=cards2.lastIndex && cards2[cards2Index] == goal[i]) {
                cards2Index++
            } else {
                return "No"
            }
        }
        return answer
    }
}